src/Conversation.tsx

```diff
@@ -7,17 +7,35 @@ import {
 } from 'openai/resources/index';
 import AssistantMessage from './AssistantMessage';
 import { Content } from './Content';
+import {
+  MessageHistoryStore,
+  useMessageHistoryStore,
+} from './stores/messageHistoryStore';
+import { TextStreamStore, useTextStreamStore } from './stores/textStreamStore';
 
 export type ChatMessage =
   | ChatCompletionMessageParam
   | ChatCompletionToolMessageParam
   | ChatCompletionAssistantMessageParam;
 
 export interface ConversationProps {
-  messages: ChatMessage[];
+  messageHistoryStore: MessageHistoryStore;
+  progressStore: TextStreamStore;
+  messageStore: TextStreamStore;
+  errorStore: TextStreamStore;
+  isRequesting: boolean;
 }
 
-const ConversationComponent = ({ messages }: ConversationProps) => {
+const ConversationComponent = ({
+  messageHistoryStore,
+  isRequesting,
+  progressStore,
+  errorStore,
+  messageStore,
```

Comment:
We chatted a bit about creating a container around this component that would subscribe to these stores, and would allow just the snapshots to be passed as props, and then we won't need to skip the unit tests.